provider/aws/lambda/lifecycle/main.go

```diff
@@ -114,6 +129,32 @@ func handle(r Record) error {
 	return nil
 }
 
+func waitForInstanceDrain(cluster, ci string) error {
+
```

Comment:
I believe we can and worth a shot. I went this route first to avoid any state changes that happen when calling the Waiter with tasks grabbed before or after instance change to `DRAINING`.